SYCL/ESIMD/api/functional/value.hpp

```diff
@@ -0,0 +1,185 @@
+To bit_cast(const From &src) noexcept {
+  To dst;
+  std::memcpy(&dst, &src, sizeof(To));
+  return dst;
+}
+
+// Initializes the sycl::half value by using two bytes given:
+//  - the higher byte, including the sign bit
+//  - the lower byte, including the part of mantissa
+//
+// This implementation doesn't depend on:
+//  - the byte order of both the unsigned types and the floating type itself;
+//  - the existence of the optional std::uint16_t type
+//  - compiler optimisations related to the strict aliasing rules
+sycl::half half_from_bytes(unsigned char hi, unsigned char lo) noexcept {
+  const size_t size = sizeof(sycl::half);
+  static_assert(CHAR_BIT == 8, "Unexpected byte size, input values may broke");
+  static_assert(size == 2, "Invalid number of bytes for half type");
+
+  const unsigned char in[size] = {lo, hi};
+  unsigned char index[size];
+  unsigned char out[size];
+
+  sycl::half indexHint = 2;
+  const unsigned char indexCoeff = 64;
+
+  memcpy(index, &indexHint, size);
+  index[0] /= indexCoeff;
+  index[1] /= indexCoeff;
+
+  assert(index[0] + index[1] == 1);
+
+  out[0] = in[index[0]];
+  out[1] = in[index[1]];
+  return esimd_test::bit_cast<sycl::half>(out);
+}
+
+// Struct that provise methods for constructing inf, lowest, nan, denorm_min and
+// max values for reference data.
+template <typename DataT> struct value {
+  static DataT inf() {
+    static_assert(type_traits::is_floating_point_v<DataT>,
+                  "Infinity required only for floating point data types.");
+
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b01111100u, 0b00000000u);
+    } else {
+      return std::numeric_limits<DataT>::infinity();
+    }
+  }
+
+  static DataT lowest() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return -half_from_bytes(0b01111011u, 0b11111111u);
+    } else {
+      return std::numeric_limits<DataT>::lowest();
+    }
+  }
+
+  static DataT denorm_min() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b00000000u, 0b00000001u);
+    } else {
+      return std::numeric_limits<RefType>::denorm_min;
+    }
+  }
+
+  static DataT nan(unsigned char opcode = 42u) {
+    static_assert(
+        type_traits::is_floating_point_v<DataT>,
+        "nan can value can retrieved only for floating point variable type.");
+    if constexpr (std::is_same_v<DataT, double>) {
+      return sycl::nan(42UL);
+    } else if constexpr (std::is_same_v<DataT, float>) {
+      return sycl::nan(42U);
+    } else if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b11111110u, 0b00000000u + opcode);
+    }
+  }
+
+  static DataT max() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b01111011u, 0b11111111u);
+    } else {
+      return std::numeric_limits<DataT>::max();
+    }
+  }
+};
+
+// Filling obtained std::vector with reference data according to current tested
```

Comment:
```suggestion
// Provides std::vector with the reference data according to the currently tested
```
Just as a suggestion )

Reply:
this comment was applied in this commit: a1b88dbf574ff747afa3a0546423fdc6ce87a587